### ISSUE TITLE
Tensor precision manifests differently between CPU and GPU.

### ISSUE THREAD

**Issue by isomatical** [pytorch/pytorch#68701]
- Device : Nvidia Jetson Xavier Nx.
- Problem: Shown in the pic below.
![image](https://user-images.githubusercontent.com/37702903/142760216-3355449f-ce5a-4b43-ab51-86ec46b96397.png)

But when tested on another device with 1080ti, it turned out normal as expected.
![image](https://user-images.githubusercontent.com/37702903/142760309-38b06656-7690-4423-8e83-d925086ee3b4.png)

**Comment by ngimel** [comment]
cc @malfet - does this jetson issue sound familiar? Who should we cc for jetson?

**Comment by malfet** [comment]
@ngimel no, as we don't do any builds for Jetson.
@ptrblck do you know who should be the right person to CC and ask to validate it against latest JetPack

**Comment by ptrblck** [comment]
Yes, please tag me and @puririshi98. I can also add us to the `module:jetson` label once I find the issue to edit.

Rishi, could you try to reproduce this behavior in the current JetPack release, please?

**Comment by puririshi98** [comment]
I used the latest unpublished 1.11 jetson wheel. It can be downloaded from [here](https://drive.google.com/file/d/1SvR2M26AXLg97MXw2J75CSmCCCffkydM/view?usp=sharing).

[Install guide](https://docs.google.com/document/d/1TaKmtna-k5aX3lor9nyPEhpCky0GnVYcPRFXQ-u05lE/edit?usp=sharing)
(follow the second option for step 3)

on a jetson tx2  
```
>>> import torch
>>> a=torch.rand(1)
  >>> a
tensor([0.9616])
>>> a.device
device(type='cpu')
>>> a=a.cuda()
>>> a.device
device(type='cuda', index=0)
>>> a
tensor([0.9616], device='cuda:0')
```

on a xavier_nx:

```
>>> import torch
>>> a=torch.rand(1)
>>> a
tensor([0.0055])
>>> a.device
device(type='cpu')
>>> a=a.cuda()
>>> a.device
device(type='cuda', index=0)
>>> a
tensor([0.0055], device='cuda:0')
```

@isomatical I cannot reproduce the issue on my end. It would be interesting to see if using the 1.11 wheel I shared on your setup fixes the issue. 
If installing the wheel I posted fixes the issue, but you would still like me to investigate why the issue existed previously; could you share the wheel of torch, its version, and the commit it is built off that you are using, so that I can compare to the latest wheel I am using? 
Otherwise, if the bug still exists on your end even with my wheel, then I would need to analyze the difference between your jetson xavier and my own setup, in which case  a list of all install libs and python modules on your machine would be useful.

**Comment by isomatical** [comment]
@puririshi98 Thanks! I tried the latest 1.10 wheel that and the problem got fixed! To investigate whether or not the problem was caused by the previous wheel (1.7.0) that I used, I reinstalled it and the same behavior occured. Then I installed and tested the latest torch1.7.0 wheel from [here](https://forums.developer.nvidia.com/t/pytorch-for-jetson-version-1-10-now-available/72048 ) and it  turned out ok. So maybe something's wrong with [that specific wheel .](https://drive.google.com/file/d/1cgpvtQQhYhsnGgcfLM7H4GSZqrxmFouh/view?usp=sharing)
